springdb/frontend/src/sections/@dashboard/app/CytoCanvas.js

```diff
@@ -0,0 +1,123 @@
+import cytoscape from 'cytoscape';
+
+import {Component} from 'react';
+
+// ----------------------------------------------------------------------
+
+export default class CytoCanvas extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {};
+    }
+    
+    render(){
+        return (
+                <div id="cyto_canvas" style={this.canvasStyle} ref={el => this.el = el} /> 
+        );
+    }
+
+    componentDidMount() {
+        // on initial mount, create a new cytoscape canvas
+        // and hook that canvas up to this element's div
+        let cytoInstance = cytoscape({
+            container: this.el,
+            style: this.initialStyles,
+        });
+
+        this.setState({
+            cytoInstance: cytoInstance
+        })
+    }
+
+    componentWillUnmount(){
+        // when this React component is unmounted, also unmount the Cytoscape canvas
+        this.state.cytoInstance.unmount();
+        this.state.cytoInstance.destroy();
+    }
+
+    componentDidUpdate() {
```

Comment:
Hmm.  I see how it works, but it seems not React-ish somehow.  The graphNodes are treated as props, but it seems like they are functioning more like state?  An alternative might be a [fully controlled component with a key](https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#recommendation-fully-uncontrolled-component-with-a-key) so that when the nodes are changed, the component is just replaced.  Then you don't have to worry about updating after a "state-like" change.